SalesforceCommerceCloud/pwa-kit
# [Storefront Preview] Avoid stale SCAPI responses (@W-15214446@)

In Storefront Preview mode, it's possible to get stale SCAPI responses, after you set the Shopper Context. You expected to see the new personalized data (e.g. a promo price), but got the older cached data instead.

One solution/workaround implemented in this PR is to set a cache breaker.

Todos:
- [ ] support the older PWA Kit v2 ← will be in another PR

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

Compare the before and after. 

Before: https://runtime.commercecloud.com/salesforce-systems/scaffold-pwa/production/preview?customQualifiers=%7B%7D&customerGroupIds=&effectiveDateTime=null&path=/global/en-GB/product/25695327M?color%3DJJI15XX%26size%3D9SM%26pid%3D701644153083M&sourceCode=
1. Note that the product price should be the regular price
2. In the Preview form, enter `tester=yes` as the custom qualifier
3. Click on the Preview button
4. Expecting the price to be the discounted promo price, but got the cached regular price instead.

After: https://runtime.commercecloud.com/mobify/blitz/will/preview?path=/global/en-GB/product/25695327M?color%3DJJI15XX%26size%3D9SM%26pid%3D701644153083M
1. Note that the product price should be the regular price
2. In the Preview form, enter `tester=yes` as the custom qualifier
3. Click on the Preview button
4. The price is immediately updated with the discounted promo price.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
